packages/cli/src/commands/deploy/index.ts

```diff
@@ -194,6 +235,21 @@ export default async (client: Client) => {
       );
       return 1;
     }
+
+    const prebuiltBuild = fs.readJsonSync(
+      join(path, '.vercel/output/builds.json'),
+      { throws: false }
+    );
+    if (prebuiltBuild && prebuiltBuild.target !== target) {
```

Comment:
Let's also handle if `builds.json` exists, but `target` was not set.

```suggestion
    if (prebuiltBuild?.target && prebuiltBuild.target !== target) {
```